### PR TITLE
[video][web] Add PiP support for web

### PR DESCRIPTION
**Pull request by behenate**
# Why

PiP is supported on Android and iOS, but not web

# How

HTML5 player has a PiP API, exposed it via our methods

# Test Plan

Tested in BareExpo in Chromium based browsers and Safari (Firefox doesn't support HTML5 PiP API)

### CODE REVIEW

File: packages/expo-video/src/VideoView.tsx [hunk 1]
```diff
@@ -7,9 +7,10 @@ import type { VideoViewProps } from './VideoView.types';
 
 /**
  * Returns whether the current device supports Picture in Picture (PiP) mode.
+ *
+ * > **Note:** All major web browsers, except Firefox, support Picture in Picture (PiP) mode.
```
Comment (amandeepmittal) [inline packages/expo-video/src/VideoView.tsx:11]:
```suggestion
 * > **Note:** All major web browsers support Picture in Picture (PiP) mode except Firefox.
```

Let's add the exception at the end so the sentence does not require any pauses.

**Review by amandeepmittal — APPROVED** [review]
Lgtm! Left one tiny suggestion for improved readability.

File: packages/expo-video/src/VideoView.web.tsx [hunk 2]
```diff
@@ -48,8 +55,36 @@ export const VideoView = forwardRef((props: { player?: VideoPlayer } & VideoView
     exitFullscreen: () => {
       document.exitFullscreen();
     },
+    startPictureInPicture: () => {
+      videoRef.current?.requestPictureInPicture();
+    },
+    stopPictureInPicture: () => {
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:61]: I think our types are not correct for these functions – they are all async on iOS, so the return type should be `Promise<void>` or, in other words, they should be async functions. Then you can use `try..catch` with `await` instead.

**Review by tsapeta — CHANGES_REQUESTED** [review]

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: packages/expo-video/src/VideoView.types.ts [hunk 3]
```diff
@@ -59,25 +59,18 @@ export interface VideoViewProps extends ViewProps {
 
   /**
    * A callback to call after the video player enters Picture in Picture (PiP) mode.
-   * @platform android
-   * @platform ios
```
Comment (Simek) [inline packages/expo-video/src/VideoView.types.ts:63]: We should add `Web` instead of removing platform tags. If tags are removed, all platforms are repeated for the package, so `tvOS` will be included on the list.

Comment (behenate) [reply packages/expo-video/src/VideoView.types.ts:63]: @Simek I totally didn't think of that! Good catch :)

File: packages/expo-video/src/VideoView.web.tsx [hunk 4]
```diff
@@ -24,6 +24,13 @@ function mapStyles(style: VideoViewProps['style']): React.CSSProperties {
   return flattenedStyles as React.CSSProperties;
 }
 
+export function isPictureInPictureSupported(): boolean {
+  const userAgent = window.navigator.userAgent;
+  // Chromium and WebKit based browsers are supported
+  // https://developer.mozilla.org/en-US/docs/Web/API/Picture-in-Picture_API#browser_compatibility
+  return !!userAgent && (userAgent.includes('Chrome') || userAgent.includes('Safari'));
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.web.tsx:31]: For sure there is another smarter way to check whether PiP is supported. Depending on user agent is almost never a good option. Maybe something like:
```suggestion
  return typeof document === 'object' && typeof document.exitPictureInPicture === 'function';
```

File: packages/expo-video/src/VideoPlayer.types.ts [hunk 5]
```diff
@@ -78,10 +78,7 @@ export declare class VideoPlayer extends SharedObject<VideoPlayerEvents> {
 
   /**
    * Boolean value indicating if the player should correct audio pitch when the playback speed changes.
-   * > On web, changing this property is not supported, the player will always correct the pitch.
    * @default true
-   * @platform android
-   * @platform ios
```
Comment (tsapeta) [inline packages/expo-video/src/VideoPlayer.types.ts:84]: Is this change intentional?

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: packages/expo-video/src/VideoView.tsx [hunk 6]
```diff
@@ -21,15 +25,15 @@ export class VideoView extends PureComponent<VideoViewProps> {
   /**
    * Enters fullscreen mode.
    */
-  enterFullscreen() {
-    this.nativeRef.current?.enterFullscreen();
+  async enterFullscreen() {
```
Comment (tsapeta) [inline packages/expo-video/src/VideoView.tsx:28]: Since the function now has a `return`, explicitly specifying the returned type might be a good idea.

File: packages/expo-video/src/VideoPlayer.types.ts (hunk 5 above)
Comment (behenate) [reply packages/expo-video/src/VideoPlayer.types.ts:84]: Yes, initially the preserve pitch didn't work on web, but then it turned out it's possible to apply it. It's supported now and the docs weren't updated apparently.

**Review by tsapeta — APPROVED** [review]
